pkg/networkpolicy/multinetegressrule.go

```diff
@@ -54,6 +54,40 @@ func (builder *EgressRuleBuilder) WithPortAndProtocol(port uint16, protocol core
 	return builder
 }
 
+// WithProtocol appends new item with only protocol to Ports list.
+func (builder *EgressRuleBuilder) WithProtocol(protocol corev1.Protocol) *EgressRuleBuilder {
+	glog.V(100).Infof("Adding protocol %s to EgressRule", protocol)
+
+	if builder.errorMsg != "" {
+		return builder
+	}
```

Comment:
could you please use the validate function (like in [mco/mcp.go](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/mco/mcp.go#L201-L203)) and place it before the glog?